### PR TITLE
container dimensions, check layout feature support

### LINKED ISSUE #844: width and height should be usable on containers
just not for dagre

### PR DESCRIPTION
**Pull request by alixander**
1. Removes restriction of adding dimensions on containers
2. Adds a check of compatibility between feature and layout. There's issue reports and confusion when we let features be used when it's not supported by the layout engine. E.g. near key being set to another object doesn't work. They think it's a bug on our side.

closes #844

### CODE REVIEW

File: d2layouts/d2elklayout/layout.go [hunk 1]
```diff
@@ -192,13 +198,18 @@ func Layout(ctx context.Context, g *d2graph.Graph, opts *ConfigurableOpts) (err
 				EdgeEdgeBetweenLayersSpacing: 50,
 				HierarchyHandling:            "INCLUDE_CHILDREN",
 				ConsiderModelOrder:           "NODES_AND_EDGES",
+				// Why is it (height, width)?
```
Comment (alixander) [inline d2layouts/d2elklayout/layout.go:201]: i have no clue. but it works.

**Comment by alixander** [comment]
(@nhooyr tagged for second point of PR -- CLI & plugin changes)

File: d2layouts/d2elklayout/layout.go [hunk 2]
```diff
@@ -192,13 +198,21 @@ func Layout(ctx context.Context, g *d2graph.Graph, opts *ConfigurableOpts) (err
 				EdgeEdgeBetweenLayersSpacing: 50,
 				HierarchyHandling:            "INCLUDE_CHILDREN",
 				ConsiderModelOrder:           "NODES_AND_EDGES",
+				// Why is it (height, width)? I have no clue, but it works.
```
Comment (nhooyr) [inline d2layouts/d2elklayout/layout.go:201]: 😂

File: d2plugin/plugin_elk.go [hunk 3]
```diff
@@ -85,8 +85,11 @@ func (p elkPlugin) Info(ctx context.Context) (*PluginInfo, error) {
 		f.AddToOpts(opts)
 	}
 	return &PluginInfo{
-		Name:      "elk",
-		Type:      "bundled",
+		Name: "elk",
+		Type: "bundled",
+		Features: map[PluginFeature]struct{}{
```
Comment (nhooyr) [inline d2plugin/plugin_elk.go:90]: This should be an array not a map, even if that makes checking for a feature a little harder.

**Review by nhooyr — CHANGES_REQUESTED** [review]

**Comment by alixander** [comment]
oh i can add https://github.com/terrastruct/d2/issues/723 to this

File: d2plugin/plugin_elk.go (hunk 3 above)
Comment (alixander) [reply d2plugin/plugin_elk.go:90]: why? the ordering does not matter.

Comment (alixander) [reply d2plugin/plugin_elk.go]: i guess if they print help and the order differs each time b/c of map iteration not being deterministic, that's bad.

**Review by nhooyr — COMMENTED** [review]
✅

**Pull request merged**